README.md

```diff
@@ -1,42 +1,132 @@
+	lb:
+	  image: 'dockercloud/haproxy:latest'
+	  links:
+	    - web
+	  ports:
+	    - '80:80'
+	  roles:
+	    - global
+
+### Running with docker legacy links
+
+Legacy link refers to the link created before docker 1.10, and the link created in default bridge network in docker 1.10 or after.
+
+##### example of legacy links using docker cli
+
+    docker run -d --name web1 dockercloud/hello-world
+    docker run -d --name web2 dockercloud/hello-world
+    docker run -d -p 80:80 --link web1:web1 --link web2:web2 dockercloud/haproxy
+
+#### example of docker-compose.yml v1 format:
+
+	web1:
+	  image: 'dockercloud/hello-world:latest'
+	web2:
+	  image: 'dockercloud/hello-world:latest'
+	lb:
+	  image: 'dockercloud/haproxy:latest'
+	  links:
+	    - web1
+	    - web2
+	  ports:
+	    - '80:80'
+
+**Note**: The link alias matters. Any link alias sharing the same prefix and followed by "-/_" with an integer is considered to be from the same service. For example: `web-1` and `web-2` belong to service `web`, `app_1` and `app_2` are from service `app`, but `app1` and `web2` are from different services.
+
+
+### Running with docker-compose v2(new links)
+
+Docker Compose 1.6 starts to support a new format of the compose file. In the new version(v2), the old link that injects environment variables is deprecated. Similar to using legacy links, here list some differences that you need to notice:
+
+- This image must be run using Docker Compose, as it relies on the Docker Compose labels for configuration
+- The container need access to the docker socket, you must mount the correct files and set the related environment to make it work
+- Although link is not needed from docker 1.10 for service discovery, you still need to set the link, so that `dockercloud/haproxy knows witch service it needs to balance. You don't need to set the linked alias, as it not used.
+- DO not overwrite `HOSTNAME` environment variable in `dockercloud/haproxy container`
+- Similar to running `dockercloud/haproxy` in Docker Cloud, auto reconfiguration is supported when the linked services scales, linked container starts/stops 
+
+##### example of docker-compose.yml running in linux:
+
+	version: '2'
+	services:
+	  web:
+	    image: dockercloud/hello-world
+	   lb:
+	    image: dockercloud/haproxy
+	    links:
+	      - web
+	    volumes:
+	      - /var/run/docker.sock:/var/run/docker.sock
+	    ports:
+	      - 80:80
+
+##### example of docker-compose.yml running in Mac OS
+
+	version: '2'
+	services:
+	  web:
+	    image: dockercloud/hello-world
+	  lb:
+	    image: dockercloud/haproxy
+	    links:
+	      - web
+	    environment:
+	      - DOCKER_TLS_VERIFY
+	      - DOCKER_HOST
+	      - DOCKER_CERT_PATH
+	    volumes:
+	      - $DOCKER_CERT_PATH:$DOCKER_CERT_PATH
+	    ports:
+	      - 80:80
+
+Once the stack is up, you can scale the web service like `docker-compose scale web=3`, and the `dockercloud/haproxy` will config itself and reload the changes accordingly.
```

Comment:
Once the stack is up, you can scale the web service using `docker-compose scale web=3`. `dockercloud/haproxy` will automatically reload its configuration.